scripts/monitor/visualization/resource_monitoring.py

```diff
@@ -0,0 +1,596 @@
+        """
+        Useful for tasks split into many shards
+        Will only plot the first disk in array, use in conjunction with disk_requested_for_task
+        :param task: task name
+        :return: plot
+        """
+        x = self.disk_requested_for_task(task).loc[:, "shard"]
+        y1 = self.disk_requested_for_task(task).loc[:, "runtime_disk_total_gb"]\
+            .map(lambda arr: arr[0]).astype("float64", copy=True)
+        y2 = self.disk_requested_for_task(task).loc[:, "meta_disk_total_gb"]\
+            .map(lambda arr: arr[0]).astype("float64", copy=True)
+        plt.plot(x, y1)
+        plt.plot(x, y2)
+        plt.legend(["Runtime Disk Usage(gb)", "Meta Disk Usage(gb)"])
+        plt.show()
+        # plt.savefig(f"{__name__}.pdf")
+        return plt.figure()
+
+    def plot_task_mem_requested(self, task: str) -> plt.Figure:
+        """
+        Useful for with tasks split into many shards
+        Use in conjunction with mem_requested_for_task
+        :param task: task name
+        :return: plot
+        """
+        x = self.mem_requested_for_task(task).loc[:, "shard"]
+        y1 = self.mem_requested_for_task(task).loc[:, "runtime_mem_total_gb"].astype("float64", copy=True)
+        y2 = self.mem_requested_for_task(task).loc[:, "meta_mem_total_gb"].astype("float64", copy=True)
+        plt.plot(x, y1)
+        plt.plot(x, y2)
+        plt.legend(["Runtime Memory Usage(gb)", "Meta Memory Usage(gb)"])
+        plt.show()
+        # plt.savefig(f"{__name__}.pdf")
+        return plt.figure()
+
+    def task_input_size(self, task_name_or_index, shard_index=None) -> float:
+        """
+        Summation of a task's input file sizes
+        Use:
+        task_input_size(row index of task/shard), or task_input_size(task name, shard index)
+        :param task_name_or_index: task name or index (details above)
+        :param shard_index: shard index if task name is given
+        :return:
+        """
+        pos = 0
+        if shard_index is not None:
+            tasks = self.meta_runtime.loc[:, "runtime_task_call_name"]  # find row index
+            shards = self.meta_runtime.loc[:, "runtime_shard"]
+            for i in range(0, tasks.size):
+                if tasks[i] == task_name_or_index and shards[i] == shard_index:
+                    pos = i
+        else:
+            if isinstance(task_name_or_index, int):
+                pos = task_name_or_index  # if only task_name is supplied it's treated as the row index
+            else:
+                raise Exception(f"Need an integer row index but received {task_name_or_index}")
+        input_ds = pd.DataFrame(data=self.meta_runtime.loc[pos, "meta_inputs"])
+        return input_ds[input_ds['type'].astype(str) == "file"].loc[:, "value"].astype(float).sum()
+
+    # ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+    def cpu_use_violin(self, instance_id: int) -> plt.Figure:
+        """
```

Comment:
It's unclear if this is intended to be used for sharded tasks or one shard of a sharded task, or both.

Reply:
It's supposed to be used for both unsharded tasks or 1 shard of a sharded task, but I just can't do the latter for the website since it's static